src/models/codex-record-metadata.js

```diff
@@ -55,13 +55,13 @@ const schema = new mongoose.Schema({
   }],
   files: [{
     ref: 'CodexRecordFile',
-    permissions: ['approved'],
     type: mongoose.Schema.Types.ObjectId,
+    permissions: ['approved-unless-historical-provenance-is-public'],
```

Comment:
I think we should come up with terminology for permissions and then document them.

e.g.,

owner (full permissions)
operator (everything but modification permissions)
viewer (owner has shared the record w/ them)
etc.

Reply:
I definitely struggled with naming this permission. Your list is essentially what I already have, just named a little differently:

1. owner - nothing is stripped from responses
1. approved - owner-only stuff is stripped from responses (`pendingUpdates` & `whitelistedAddresses`)
1. logged-in - provenance is stripped from responses
1. no permissions - the field is not stripped for anyone

The problem is this new permission doesn't really fit into any of those categories, since it's kind of a conditional permission (i.e. "approved" if the Record is private, but "no permissions" if it's public.)

The way this "permissions" stuff works is definitely a little confusing. It's not really "permissions" so much as it is "categories for what should be removed when sending Codex Records in responses" if that makes sense... I agree it could be a little more intuitive for sure.